substrate-archive/src/types.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright 2017-2019 Parity Technologies (UK) Ltd.
+// This file is part of substrate-archive.
+
+// substrate-archive is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// substrate-archive is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with substrate-archive.  If not, see <http://www.gnu.org/licenses/>.
+
+use sp_runtime::traits::Block as BlockT;
+use substrate_archive_common::error::Result;
+
+pub trait ThreadPool: Send + Sync {
```

Comment:
Not from this PR, but we can just get rid of the `ThreadPool` trait, it's not being used for anything anymore